### PR TITLE
Fixes #18: Merge work on use of glossaries-extra for notation & abbrevs

### LINKED ISSUE #18: Use one of the glossaries packages to create the "abbreviations" section
At the moment, the "abbreviations" section is generated completely manually. It would be nice to generate it automatically, along with the ability to list the page of first appearance and other nifty features which packages like [`glossaries`](https://ctan.org/pkg/glossaries) and [`nomencl`](https://ctan.org/pkg/nomencl) offer.

### PR DESCRIPTION
**Pull request by eyalroz**
Fixes #18. See discussion in #42 .

### CODE REVIEW

File: front/abbrevs.tex [hunk 1]
```diff
@@ -1,9 +1,79 @@
...
+
+\newacronym[%
+  category=notation-category,%
+description=A technology used in data storage devices]%
+  {smart}{SMART}{Self-Monitoring, Analysis and Reporting Technology}
+
+\newacronym[%
+  category=notation-category,%
+  description=to build or produce something rather than purchasing it ready-made by others{,} or paying someone else to make it]%
+  {DIY}{DIY}{Do it yourself}
```
Comment (haggaie) [inline front/abbrevs.tex:36]: A minor comment: I think you don't need capital D in Do. According to [Grammarly](https://www.grammarly.com/blog/abbreviations/), when the words of an acronym are presented they only need to be capitalized if they form a proper noun.

File: front/abbrevs.tex [hunk 2]
```diff
@@ -1,9 +1,79 @@
...
+% The following will alter the style for typesetting abbreviations when using 
+% the \gls command. Note you can also use multiple styles by categorizing 
+% abbreviations; see the documentation for the glossaries-extras package at:
+% https://ctan.org/pkg/glossaries-extra
+%
+%\setabbreviationstyle[acronym]{long-short-sc}
+%
+% If you're wondering why we're setting the seemingly-redundant "notation 
+% category", that's a hack discussed here:
+% https://tex.stackexchange.com/q/630541/5640
```
Comment (haggaie) [inline front/abbrevs.tex:19]: Perhaps it is also good to mention here that without the category, all pages where the term appear would be indexed?

File: thesis.tex [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
 %   defended - drop this option.
-% noabbrevs          no notation & abbreviations list will be included
-%   in the thesis.
+% noabbrevs          avoid generation of a notation & abbreviations list
 %
 % Additionally, you must specify the degree for which you're writing
 % your thesis (MSc/PhD/MArch etc.)
 %
-\documentclass[PhD]{misc/iitthesis}
+\documentclass[PhD,fullpageDraft]{misc/iitthesis}
```
Comment (haggaie) [inline thesis.tex:17]: Are you sure you wanted this change to be in the master branch? It doesn't seem related to glossary support.

File: front/abbrevs.tex [hunk 4]
```diff
@@ -1,9 +1,79 @@
...
+  category=notation-category,%
+  description=a four-letter acronym]
+  {etla}
+  {ETLA}
+  {extended three-letter acronym}
+
+\newabbreviation[%
+  category=notation-category,%
+  description=]% This abbreviation has no description; only the abbreviation and the unabbreviated form will be shown
+  {aut}{Aut}{Automorphism group}
```
Comment (haggaie) [inline front/abbrevs.tex:48]: This abbreviation does not have `type=notation`, and doesn't appear in the result when using `\keepUnusedAbbreviations`. Did you mean it to be an example of multiple glossaries, or is it just a mistake?

File: .latexmkrc [hunk 5]
```diff
@@ -0,0 +1,24 @@
+add_cus_dep( 'acn', 'acr', 0, 'run_makeglossaries' );
+add_cus_dep( 'glo', 'gls', 0, 'run_makeglossaries' );
+add_cus_dep( 'ntn', 'not', 0, 'run_makeglossaries' );
+
+push @generated_exts, 'glo', 'gls', 'glg';
+push @generated_exts, 'acn', 'acr', 'alg';
```
Comment (haggaie) [inline .latexmkrc:6]: I saw glo/gls files are still generated (due to the automorphism example), but are acr/acn files supposed to be generated too?

**Review by haggaie — COMMENTED** [review]
Hi, I tried this with and without \keepUnusedAbbreviations, and it seems to be working fine. I wrote a few comments inline. I haven't tested it with multiple glossaries though.

File: front/abbrevs.tex (hunk 4 above)
Comment (eyalroz) [reply front/abbrevs.tex:48]: Just a mistake.

File: thesis.tex (hunk 3 above)
Comment (eyalroz) [reply thesis.tex:17]: No, will revert this.

File: .latexmkrc (hunk 5 above)
Comment (eyalroz) [reply .latexmkrc:6]: No, but - just in case.
